drud/ddev
# Allow override of mysql and php configuration, fixes #654

## The Problem/Issue/Bug:

Users need to be able to override php and mysql configuration in the same way as they can override nginx configuration.

## How this PR Solves The Problem:

* Update docker-compose.yml to mount .ddev as /ddev_config in  both web and db containers. This allows those containers to easily change their behavior based on the contents of /ddev_config.
* Remove an obsolete section that created an nginx-site.conf for nginx. After https://github.com/drud/ddev/pull/669 went in this is totally obscene :)

## Manual Testing Instructions:

The web and db images have been updated in this PR. 

* Delete .ddev and reconfigure
* Add a php.ini and my.cnf to your heart's delight
* ddev start
* Verify that the site admin/reports/status/php shows the changed value
* Verify that the mysql behavior changes. A nice test is to fix the utf8mb4 problem.

## Automated Testing Overview:

Tests are added in both containers for the new behavior. Since the behavior is essentially all in the containers that's probably good enough.

## Related Issue Link(s):

Mariadb container change: https://github.com/drud/mariadb-local/pull/46
Nginx container change: https://github.com/drud/docker.nginx-php-fpm-local/pull/54
OP: #654 - need to override my.cnf


OP: #654 problems with importing mysql databases.

## Release/Deployment notes:
